src/EventListener/PurchaseSubscriber.php

```diff
@@ -64,26 +56,18 @@ public static function getSubscribedEvents(): array
 
     public function track(ResourceControllerEvent $event): void
     {
+        /** @var OrderInterface|mixed $order */
         $order = $event->getSubject();
-
-        if (!$order instanceof OrderInterface) {
-            return;
-        }
+        Assert::isInstanceOf($order, OrderInterface::class, sprintf(
+            'You must implement the %s in your Sylius application. Read the readme here: https://github.com/Setono/SyliusGoogleAdsPlugin',
+            OrderInterface::class
+        ));
 
         $channel = $order->getChannel();
         if (null === $channel) {
             return;
         }
 
```

Comment:
Should it not check for the presence of the click id on the order, before creating the conversion?